src/throttlePromise.ts

```diff
@@ -1,73 +1,76 @@
-import type { ThrottleFn } from './interfaces';
+import type { ISbThrottle, Queue } from './interfaces';
 
-interface Shifted {
-  args: any;
-  self: any;
-  resolve: (args: any) => any;
+class AbortError extends Error {
+  constructor(msg: string) {
+    super(msg);
+    this.name = 'AbortError';
+  }
 }
 
-interface Queue {
-  resolve: (args: any) => any;
-  reject: (args: any) => any;
-  args: any[];
-  self: any;
-}
+function isFinite(value: number) {
```

Comment:
do you find any specific reason why we should keep this function and not using `Number.isFinite()` instead? I guess the only reason would be type coercion but here we are using it only with `throttledQueue` and the type is already coerced

Reply:
Great point. It was already there, but good time to remove it 😉